docs/resource-schemas.md

```diff
@@ -336,18 +336,16 @@ have the following values:
 
 Asset docs is used to determine which supporting documentation is available to fetch and render as
 page tabs in the asset details for the asset as well as its location. For each optional document
-key, a path should be included; The path **must** be either a Github URL according to the
-`https://{host}/{org}/{repo}/blob/{branch}/{path-to-file}` format, or a path relative to the
-carbon.yml file location.
+key, a path should be included.
 
 Currently supported document keys are:
 
-| Document            | Description                                                       |
-| ------------------- | ----------------------------------------------------------------- |
-| `usagePath`         | When to use the asset and how it behaves.                         |
-| `stylePath`         | Asset structure and design specifications.                        |
-| `codePath`          | Asset API documentation (or where to find it).                    |
-| `accessibilityPath` | Asset accessibility considerations when designing and developing. |
+| Document            | Description                                                       | Required | Type   | Inheritable | Default | Valid values                                                                                                                                                            |
+| ------------------- | ----------------------------------------------------------------- | -------- | ------ | ----------- | ------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `usagePath`         | When to use the asset and how it behaves.                         | Optional | String | Yes         | -       | <ul><li>Github URL according to the `https://{host}/{org}/{repo}/blob/{branch}/{path-to-file}` format </li> <li>Path relative to the carbon.yml file location</li></ul> |
+| `stylePath`         | Asset structure and design specifications.                        | Optional | String | Yes         | -       | <ul><li>Github URL according to the `https://{host}/{org}/{repo}/blob/{branch}/{path-to-file}` format </li> <li>Path relative to the carbon.yml file location</li></ul> |
+| `codePath`          | Asset API documentation (or where to find it).                    | Optional | String | Yes         | -       | <ul><li>Github URL according to the `https://{host}/{org}/{repo}/blob/{branch}/{path-to-file}` format </li> <li>Path relative to the carbon.yml file location</li></ul> |
+| `accessibilityPath` | Asset accessibility considerations when designing and developing. | Optional | String | Yes         | -       | <ul><li>Github URL according to the `https://{host}/{org}/{repo}/blob/{branch}/{path-to-file}` format </li> <li>Path relative to the carbon.yml file location</li></ul> |
```

Comment:
In order for these to be inheritable, they should have a default value. Otherwise, for the root-most asset in the inheritance chain, the value could be undefined. (since these are also marked as optional). Would it make sense for the default to be something like `./usage.mdx` (etc... for the others)?

This won't handle the case of a top-level carbon.yml defining all of the assets. @mattrosno Maybe that should push us in the direction of needing one carbon.yml per asset. Otherwise, we'll lose the association between an asset's source, its doc files, and its schema definition, which kind of goes against our co-location of assets and docs mentality.

Reply:
@jdharvey-ibm here's an example written out that explains what we're after.

Carbon Styles library has Accordion, which has usage, style, code and accessibility documentation.

The Carbon React library inherits Carbon Styles. Carbon React also has an Accordion (that matches the Carbon Styles Accordion id.) Because the documentation pages are inherited, the Carbon React Accordion doesn’t need to specify anything as it will inherit the documentation pages from Carbon Styles.

That means the Carbon React Accordion doesn't need its own documentation. If inheriting the latest Carbon Styles version, the Carbon Styles Accordion usage page will be the same as the Carbon React Accordion usage page.

But, per our inheritance specification, it's possible to inherit a specific version to "pin" it. https://github.com/carbon-design-system/carbon-platform/blob/main/docs/resource-schemas.md#library-inheritance

So if Carbon Styles has a v1 and v2, and Carbon React is inheriting Carbon Styles v1, then when viewing the latest version of everything Carbon Styles Accordion will show v2 usage content and the Carbon React Accordion will show v1 usage content.

This becomes important when for example a feature was added (and documented) in Carbon Styles v2. If Carbon React hasn't implemented that new feature yet and is still inheriting v1, you don't want to see the documentation for that v2 feature in the Carbon React docs.

Reply:
@jdharvey-ibm I typed out a lot there but glad that made sense.

Per https://github.com/carbon-design-system/carbon-platform/issues/485 (specifically proposal 3 that we landed on), if an asset doesn't specify any documentation pages, it only has the Overview tab. If it only has usage documentation for example, it only has Overview and Usage tabs.

If there's no code documentation, we won't want to be rendering a Code page tab with empty content.

Most of the PALs, and our universal patterns, only have usage documentation so a good number of our asset detail pages will have one or two page tabs for starters.

Reply:
@mattrosno Ohhh okay got it. I was thinking that we'd be enforcing stuff like accessibility compliance by requiring those pages, but it sounds like that's not the case.

Given that, it seems acceptable to allow these to remain blank all the way up to the root-most components, but I'd still argue that we could throw some default values in there and try to fetch those during indexing. If they're there, awesome! We use them without any additional config from the user. If they don't, no problem! We just hide the tab.

Reply:
I'm game for defaults and error handling that we'll need regardless.

Are you saying we support `usage.mdx`, `code.mdx`, `style.mdx`, `accessibility.mdx` where if they exist in the same directory, then we don't need to add `usagePath`, etc. to the indexing? If so, we can only support those defaults if we use `carbon.yml` for each asset?

That seems reasonable to me. Us using `carbon.yml` for an entire library seems like a short term thing. That file's going to get too big to manage. I think we'll eventually want a Carbon config file for the library and per asset in subdirectories.

Reply:
I don't think defaults make sense if assets can be an array in our schema, are we making a decision to move back to having separate files/schemas for assets and libraries? if so then we can go with the defaults from my perspective

Reply:
It'd be nice for us to keep support so a library can have one or many config files. I think it's reasonable to only support asset docs defaults if the Carbon config files are used on a per-asset basis?

Reply:
We can definitely do both in the json schema.

https://json-schema.org/understanding-json-schema/structuring.html#ref

If the definition of an asset in the json schema is put into a "$def", it can be used in multiple places and extended. This is probably what we want to do with an asset if there's two different use cases for it in the schema. For example:

```json
"properties": {
  "myThing": {
    "$ref": "#/$defs/asset"
  },
  "myOverriddenThing": {
    "$ref": "#/$defs/asset",
    "properties": {
      "assetUsage": { "default": "wow a new default" }
    },
    "required": ["assetUsage"]
  }
}

"$defs": {
  "asset": {
    "type": "object",
    "properties": {
      "assetUsage": { "type": "string", "default": "usage.mdx" },
      "otherThing": { "type": "number", "default": 123 }
    },
    "required": ["assetUsage", "otherThing"]
  }
}
```

https://user-images.githubusercontent.com/51208233/172901257-6a9f724c-6294-4b45-895d-9fd3dc5bb03b.mov


So that could allow defaults to be used in one place and not another. We'd need a way to distinguish between the two though. For now, I think the defaults could exist in a multi-asset carbon.yml file and they won't hurt anything; they'll just point to files that don't exist, so we'd want to fall back to the inherited value or if none is found, don't show the tab.

Something to consider (outside the scope of this issue): are we moving away from an assets L0 (at least back-end-wise)? If so, it probably doesn't make sense for the schema definition to have an assets L0; it'd be better suited inside of the library object itself. That would enable us to do a top-level `asset` key for the one-carbon.yml-per-asset model, and `libraries.assets` as the multiple-assets-per-carbon.yml model, and then enforce that you only specify one or the other of those two.

Reply:
^^ sound good to me, added defaults! I think if we-re going the one-asset-per-carbon.yml way it'd probably make morse sense to have a separate carbon-asset.yml file like we used to with a separate asset schema instead of having a top-level `asset` ket on top of `library.assets` within the same file schema, do we need to open a separate ticket to discuss this further?

Reply:
Yeah, let's definitely open a discussion/investigation ticket for that to get it on the books.

Reply:
@jdharvey-ibm @francinelucca a constraint to keep in mind (just a reminder) - it's common for some libraries to have a components folder where each component has its own JS file like `a-component.js` and `another-component.js` to both live in the `/components` directory.

We don't want to require libraries to move files around so it's "one asset per directory". That's why our current `carbon.yml` structure works well to handle this scenario of multiple assets in same directory.

Reply:
Created #852 and copied some of these comments over 👯 . @mattrosno does this one look good to go?

Reply:
@mattrosno In that situation you outlined, what's the typical strategy employed to distinguish the usage.mdx page for one component from another? Are they naming their mdx files in a way that somehow matches their component name?

Reply:
I'm stating for a simple component that doesn't have a `my-component.module.scss` or a `my-component.stories.js` or tests or anything... if a component only lives in a single JS file, it's common for all of those components to live together in one directory.

In that scenario, we'd need to be able to index all of those components with one `carbon.yml` file.

The moment a component has a styles file, or stories, or tests, or documentation (like `usage.mdx`), at that point the component often lives in its own directory so it'd be natural to have the `carbon.yml` in that directory that only indexes that component.